### ISSUE TITLE
Confirm the training is correct

### ISSUE THREAD

**Issue by github-luffy** [hkchengrex/CascadePSP#21]
Training my own dataset，one epoch result as below:
It     1450 [TRAIN] [grad_loss          ]: 0.0108861
It     1450 [TRAIN] [iou/orig_i         ]: 15341.660
It     1450 [TRAIN] [iou/orig_u         ]: 30589.000
It     1450 [TRAIN] [iou/new_i_224      ]: 0.0000000
It     1450 [TRAIN] [iou/new_u_224      ]: 17783.060
It     1450 [TRAIN] [iou/new_i_56       ]: 0.0000000
It     1450 [TRAIN] [iou/new_u_56       ]: 17783.060
It     1450 [TRAIN] [iou/new_i_28       ]: 0.0000000
It     1450 [TRAIN] [iou/new_u_28       ]: 17783.060
It     1450 [TRAIN] [iou/new_i_28_2     ]: 0.0000000
It     1450 [TRAIN] [iou/new_u_28_2     ]: 17783.060
It     1450 [TRAIN] [iou/new_i_28_3     ]: 0.0000000
It     1450 [TRAIN] [iou/new_u_28_3     ]: 17783.060
It     1450 [TRAIN] [iou/new_i_56_2     ]: 0.0000000
It     1450 [TRAIN] [iou/new_u_56_2     ]: 17783.060
It     1450 [TRAIN] [total_loss         ]: 0.4961829
It     1450 [TRAIN] [ce_loss/s_0        ]: 0.3328005
It     1450 [TRAIN] [l1_loss/s_0        ]: 0.0295411
It     1450 [TRAIN] [l2_loss/s_0        ]: 0.0295335
It     1450 [TRAIN] [loss/s_0           ]: 0.1135052
It     1450 [TRAIN] [ce_loss/s_1        ]: 0.0864096
It     1450 [TRAIN] [l1_loss/s_1        ]: 0.0507903
It     1450 [TRAIN] [l2_loss/s_1        ]: 0.0253867
It     1450 [TRAIN] [loss/s_1           ]: 0.0864096
It     1450 [TRAIN] [ce_loss/s_2        ]: 0.0875118
It     1450 [TRAIN] [l1_loss/s_2        ]: 0.0458868
It     1450 [TRAIN] [l2_loss/s_2        ]: 0.0255962
It     1450 [TRAIN] [loss/s_2           ]: 0.0616267
It     1450 [TRAIN] [ce_loss/s_3        ]: 0.0865094
It     1450 [TRAIN] [l1_loss/s_3        ]: 0.0508352
It     1450 [TRAIN] [l2_loss/s_3        ]: 0.0253933
It     1450 [TRAIN] [loss/s_3           ]: 0.0865094
It     1450 [TRAIN] [ce_loss/s_4        ]: 0.0865065
It     1450 [TRAIN] [l1_loss/s_4        ]: 0.0508407
It     1450 [TRAIN] [l2_loss/s_4        ]: 0.0253932
It     1450 [TRAIN] [loss/s_4           ]: 0.0865065
It     1450 [TRAIN] [ce_loss/s_5        ]: 0.0875015
It     1450 [TRAIN] [l1_loss/s_5        ]: 0.0459030
It     1450 [TRAIN] [l2_loss/s_5        ]: 0.0255950
It     1450 [TRAIN] [loss/s_5           ]: 0.0616252
It     1450 [TRAIN] [iou/orig_iou       ]: 0.5015417
It     1450 [TRAIN] [iou/new_iou_224    ]: 0.0000000
It     1450 [TRAIN] [iou/iou_gain_224   ]: -0.501541
It     1450 [TRAIN] [iou/new_iou_56     ]: 0.0000000
It     1450 [TRAIN] [iou/iou_gain_56    ]: -0.501541
It     1450 [TRAIN] [iou/new_iou_28     ]: 0.0000000
It     1450 [TRAIN] [iou/iou_gain_28    ]: -0.501541
It     1450 [TRAIN] [iou/new_iou_28_2   ]: 0.0000000
It     1450 [TRAIN] [iou/iou_gain_28_2  ]: -0.501541
It     1450 [TRAIN] [iou/new_iou_28_3   ]: 0.0000000
It     1450 [TRAIN] [iou/iou_gain_28_3  ]: -0.501541
It     1450 [TRAIN] [iou/new_iou_56_2   ]: 0.0000000
It     1450 [TRAIN] [iou/iou_gain_56_2  ]: -0.501541
I am not sure if it is normal, waiting for your reply.

**Comment by hkchengrex** [comment]
Doesn't look good if IOU is still at 0 after 1400 iterations. How about the tensorboard output?

**Comment by github-luffy** [comment]
I think the problem may be the groundtruth(0 and 255), groundtruth must be 0 and 1?

**Comment by hkchengrex** [comment]
The image should be in (0-255), that's just how we (and others) store the masks. The dataloader should have converted them to the proper range.

**Comment by github-luffy** [comment]
I found the groundtruth is into [0, 1),  and can you tell me why?
self.gt_transform = transforms.Compose([
            transforms.ToTensor(),
        ])

**Comment by hkchengrex** [comment]
What do you mean by why? This converts raw image (0\~255) to tensors (0\~1).
